src/Background/stat_estimators.jl

```diff
@@ -1,5 +1,43 @@
 using Statistics
 
+"""
+    sigma_clip(data, sigma_low, sigma_high; center=median, std=std, iterations=5)
+    sigma_clip(data, sigma; center=median, std=std, iteration=5)
+
+This function returns sigma clipped values of the input `data`.
+`sigma_high` and `sigma_low` are for un-symmetrical clipping, when `sigma_low = sigma_high` then they can be passed as `sigma`.
+`center`, `std` and `iterations` are optional parameters which are functions for finding central element, standard deviation and number of iterations of sigma
+clipping to be carried out.
+
+# Example
+```jldoctest
+julia> data = [1, 2, 3];
+
+julia> sigma_clip(data, 1)
+3-element Array{Float64,1}:
+ 1.0
+ 2.0
+ 3.0
+
+julia> sigma_clip(data, 1, 1)
+3-element Array{Float64,1}:
+ 1.0
+ 2.0
+ 3.0
+```
+"""
+
+function sigma_clip(data::AbstractArray, sigma_low::Number, sigma_high::Number; iterations::Number=5, center=median, std=std)
+    for i=1:iterations
+        mean = center(data)
+        deviation = std(data)
+        data = clamp.(data, mean - sigma_low * deviation, mean + sigma_high * deviation)
```

Comment:
If we want this function to be in-place I think this should become
```suggestion
        clamp!.(data, mean - sigma_low * deviation, mean + sigma_high * deviation)
```

Reply:
After making the following change, I ran the test as follows
`x = [1,2,3,4,5,6]`
`sigma_clip(x,1,1)`
It throws an error, error being:
`MethodError: no method matching clamp!(::Int64, ::Float64, ::Float64)`

Reply:
I think there shouldn't be that `.` after `clamp!` :slightly_smiling_face:

Reply:
Uh, yeah, sorry, remove the dot.  The first argument has to be an array